source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/Proxies/MethodInvocationHookProxyDecorator.cs

```diff
@@ -4,60 +4,45 @@
 
 namespace Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators.Proxies
 {
-    public delegate Task MethodInvocationHook<in TService>(TService service);
+    public delegate Task PreMethodInvocationHook<in TService, in TRequest>(TService service, TRequest request);
+    public delegate Task PostMethodInvocationHook<in TService, in TResponse>(TService service, TResponse response);
 
-    public class MethodInvocationHookProxyDecorator<TService> : ServiceProxy where TService : class
+    public class MethodInvocationHookProxyDecorator<TService,TRequest, TResponse> : ServiceProxy where TService : class
     {
-        MethodInvocationHook<TService>? preInvocation;
-        MethodInvocationHook<TService>? postInvocation;
+        PreMethodInvocationHook<TService, TRequest>? preInvocation;
+        PostMethodInvocationHook<TService, TResponse>? postInvocation;
         string methodName;
 
-        void Configure(string methodName, MethodInvocationHook<TService>? preInvocation, MethodInvocationHook<TService>? postInvocation)
+        void Configure(string methodName, PreMethodInvocationHook<TService, TRequest>? preInvocation, PostMethodInvocationHook<TService, TResponse>? postInvocation)
         {
             this.preInvocation = preInvocation;
             this.postInvocation = postInvocation;
             this.methodName = methodName;
         }
 
-        public static TService Create(TService targetService, string methodName, MethodInvocationHook<TService>? preInvocation, MethodInvocationHook<TService>? postInvocation)
+        public static TService Create(TService targetService, string methodName, PreMethodInvocationHook<TService, TRequest>? preInvocation, PostMethodInvocationHook<TService, TResponse>? postInvocation)
```

Comment:
Some tests needed the request object and some needed the response object, so I've included them in the invocation hook definitions